README.md

```diff
@@ -506,342 +557,154 @@ zenstruck_foundry:
         service: my_instantiator # your own invokable service for complete control
 ```
 
-#### Immutable
+### Initialization
 
-Factory objects are immutable:
+You can override your model factory's `initialize()` method to add default state/logic:
 
 ```php
-use App\Post;
-use Zenstruck\Foundry\Factory;
+namespace App\Factory;
+
+use App\Entity\Post;
+use Zenstruck\Foundry\ModelFactory;
+
+final class PostFactory extends ModelFactory
+{
+    // ...
+
+    protected function initialize(): self
+    {
+        return $this
+            ->published() // published by default
+            ->instantiateWith(function (array $attributes) {
+                return new Post(); // custom instantiation for this factory
+            })
+            ->afterPersist(function () {}) // default event for this factory
+        ; 
+    }
```

Comment:
I only have used this a handful of times in a *real app* but if I add the default, I could add the `self` typehint to avoid problems with the user not returning `$this`. WDYT?

Reply:
I kinda like it! These factory classes are small anyways, so adding one other empty method seems ok to me :)

Reply:
7425caf6e284525fb83c40a76cfac48cb11bfac1 & b657e07e76e4e1f8f87be35e8d055afb836e3868